### PR TITLE
Added check to object before calling its property.

### PR DESCRIPTION
**Pull request by ctam**
Fixing #22.

### CODE REVIEW

File: locallib.php [hunk 1]
```diff
@@ -126,13 +126,13 @@ function enrol_ilios_sync(progress_trace $trace, $courseid = NULL) {
                     $ue = $DB->get_record('user_enrolments', array('enrolid' => $instance->id, 'userid' => $userid));
 
                     // Continue if already enrolled with active status
-                    if (!empty($ue) && $status === (int)$ue->status) {
+                    if (!empty($ue) && $status === $ue->status) {
```
Comment (stopfstedt) [inline locallib.php]: still need to cast to int here. since, `$status` is an integer, and `$ue->status` is a string.

File: locallib.php [hunk 2]
```diff
@@ -126,13 +126,13 @@ function enrol_ilios_sync(progress_trace $trace, $courseid = NULL) {
...
                         continue;
                     }
 
                     // Enroll user
                     $plugin->enrol_user($instance, $userid, $instance->roleid, 0, 0, $status);
-                    if ($status !== (int)$ue->status) {
+                    if (!empty($ue) && $status !== $ue->status) {
```
Comment (stopfstedt) [inline locallib.php]: `$ue->status` needs casting to integer, see comment above.

File: locallib.php (hunk 1 above)
Comment (ctam) [reply locallib.php]: I believe it's an `int` from what I can see in the [database definition](https://github.com/moodle/moodle/blob/MOODLE_33_STABLE/lib/db/install.xml#L263), unless `$DB->get_record` has casted it into a string.

Comment (stopfstedt) [reply locallib.php]: selection_494

Comment (ctam) [reply locallib.php]: And yes, looks like `$DB->get_record` does cast everything to a string.  :(

**Pull request merged**